### PR TITLE
Cluster firmware inventory.

### PR DESCRIPTION
**Pull request by prasannajayabal**

### CODE REVIEW

File: Python/omevv/v1/scripts/getFirmwareInventory.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+from omevv_apis_client.models import GroupFirmwareInventoryCollectionModel
+from omevv_apis_client.models import ErrorObject
+from omevv_apis_client.models import Credential
+import constants
+import base64
+
+from omevv_apis_client.types import Response
+
+
+class ManageHostWrapper:
```
Comment (pavithra-mahadev) [inline Python/omevv/v1/scripts/getFirmwareInventory.py]: Can we rename the file as GroupFirmwareInventory?

File: Python/omevv/v1/scripts/getFirmwareInventory.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    PARSER.add_argument("--omevv_group_id", "-g", required=True, default=None, help="omevv_group_id")
+    ARGS = PARSER.parse_args()
+
+    if ARGS.ip is not None and ARGS is not None and ARGS.vcpassword is not None and ARGS.vcUUID is not None \
+            and ARGS.omevv_group_id is not None:
+        base_url = 'https://{ip}/omevv/GatewayService/v1/'.format(ip=ARGS.ip)
+        credential = Credential(username=ARGS.vcusername, password=ARGS.vcpassword)
+        output = ManageHostWrapper(base_url=base_url, vcUsercredential=credential, vCenterUUID=ARGS.vcUUID,
+                                   omevv_Group_id=ARGS.omevv_group_id).get_managed_hosts_firmware_inventory_by_group()
+        print("Output here-----------", output)
```
Comment (pavithra-mahadev) [inline Python/omevv/v1/scripts/getFirmwareInventory.py]: Can we write to a csv file, if cluster contains more no. of hosts, data will be huge

File: Python/omevv/v1/scripts/getFirmwareInventory.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    PARSER = argparse.ArgumentParser(description=__doc__,
+                                     formatter_class=argparse.RawTextHelpFormatter)
+    PARSER.add_argument("--ip", "-i", required=True, help="OME Appliance IP")
+    PARSER.add_argument("--vcusername", "-u", required=True,
+                        help="username of vcenter")
+    PARSER.add_argument("--vcpassword", "-p", required=True,
+                        help="password of vcenter")
+    PARSER.add_argument("--vcUUID", "-d", required=True, default=None,
+                        help="UUID of the relevant vCenter")
+    PARSER.add_argument("--omevv_group_id", "-g", required=True, default=None, help="omevv_group_id")
```
Comment (pavithra-mahadev) [inline Python/omevv/v1/scripts/getFirmwareInventory.py]: can we have proper help message for group id and mention it should be only cluster id.

**Review by pavithra-mahadev — CHANGES_REQUESTED** [review]

File: Python/omevv/v1/scripts/getFirmwareInventory.py (hunk 1 above)
Comment (prasannajayabal) [reply Python/omevv/v1/scripts/getFirmwareInventory.py]: @pavithra-mahadev : File name modified

File: Python/omevv/v1/scripts/getFirmwareInventory.py (hunk 2 above)
Comment (prasannajayabal) [reply Python/omevv/v1/scripts/getFirmwareInventory.py]: @pavithra-mahadev : Used common module to export csv.

File: Python/omevv/v1/scripts/getFirmwareInventory.py (hunk 3 above)
Comment (prasannajayabal) [reply Python/omevv/v1/scripts/getFirmwareInventory.py]: @pavithra-mahadev : Help updated.

File: Python/omevv/v1/scripts/GroupFirmwareInventory.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+                        help="username of vcenter")
+    PARSER.add_argument("--vcpassword", "-p", required=True,
+                        help="password of vcenter")
+    PARSER.add_argument("--vcUUID", "-d", required=True, default=None,
+                        help="UUID of the relevant vCenter")
+    PARSER.add_argument("--omevv_group_id", "-g", required=True, default=None,
+                        help="OMEVV group ID of the cluster, only group ID will be accepted")
+    ARGS = PARSER.parse_args()
+
+    if ARGS.ip is not None and ARGS is not None and ARGS.vcpassword is not None and ARGS.vcUUID is not None \
```
Comment (pavithra-mahadev) [inline Python/omevv/v1/scripts/GroupFirmwareInventory.py]: is it only ARGS? looks like it should be ARGS.vcusername

**Review by pavithra-mahadev — CHANGES_REQUESTED** [review]

File: Python/omevv/v1/scripts/GroupFirmwareInventory.py [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+import argparse
+from omevv_apis_client import AuthenticatedClient
+from omevv_apis_client.api.group_inventory import get_managed_hosts_firmware_inventory_by_group
+from omevv_apis_client.models import ManagedHost
+from omevv_apis_client.models import GroupFirmwareInventoryCollectionModel
+from omevv_apis_client.models import ErrorObject
+from omevv_apis_client.models import Credential
+import constants
+import base64
+import pandas as pd
```
Comment (pavithra-mahadev) [inline Python/omevv/v1/scripts/GroupFirmwareInventory.py]: why do we need this?

File: Python/omevv/v1/scripts/GroupFirmwareInventory.py (hunk 4 above)
Comment (prasannajayabal) [reply Python/omevv/v1/scripts/GroupFirmwareInventory.py]: @pavithra-mahadev : Corrected the typo.

File: Python/omevv/v1/scripts/GroupFirmwareInventory.py (hunk 5 above)
Comment (prasannajayabal) [reply Python/omevv/v1/scripts/GroupFirmwareInventory.py]: @pavithra-mahadev : Removed the imports not using it in the scripts.

**Review by pavithra-mahadev — APPROVED** [review]

**Review by Abhishek14-Dell — APPROVED** [review]

**Review by pavithra-mahadev — APPROVED** [review]

**Pull request merged**